### PR TITLE
Add DataFrame.append

### PR DESCRIPTION
**Pull request by sinhrks**
Almost the same as ``concat``, but not have ``interleave_partitions`` option because (personally feel) ``append`` emphasizes the order.

```
import pandas as pd
import dask.dataframe as dd

df1 = pd.DataFrame({'a': [0, 0, 0, 0], 'b': [1, 1, 1, 1]})
df2 = pd.DataFrame({'a': [2, 2, 2, 2], 'b': [3, 3, 3, 3]}, index=[4, 5, 6, 7])

ddf1 = dd.from_pandas(df1, 2)
ddf2 = dd.from_pandas(df2, 2)

ddf1.append(ddf2)
# dd.DataFrame<dataframe--append--7d332e76cf7871ad19d744929d5dbc77, divisions=(0, 2, 4, 6, 7)>

ddf1.append(ddf2).compute()
#    a  b
# 0  0  1
# 1  0  1
# 2  0  1
# 3  0  1
# 4  2  3
# 5  2  3
# 6  2  3
# 7  2  3
```

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -763,6 +763,31 @@ def mask(self, cond, other=np.nan):
...
+            from .io import from_pandas
+            other = from_pandas(other, 1)
+
+        from .multi import _append
+        if self.known_divisions and other.known_divisions:
+            if self.divisions[-1] < other.divisions[0]:
+                divisions = self.divisions[:-1] + other.divisions
+                return _append(self, other, divisions)
+            else:
+                msg = 'Unable to append 2 known divisions which is not ordered'
```
Comment (mrocklin) [inline dask/dataframe/core.py]: If we wanted to be more explicit here we could use the following message.

> Unable to append two dataframes to each other with known divisions if those divisions are not ordered.  The divisions/index of the second dataframe must be greater than the divisions/index of the first dataframe.

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -1413,6 +1438,17 @@ def join(self, other, on=None, how='left',
...
                      npartitions=npartitions)
 
+    @derived_from(pd.DataFrame)
+    def append(self, other):
+        if isinstance(other, Series):
+            msg = ('Unable to appending dd.Series to dd.DataFrame.'
+                   'Use pd.Series to append as row.')
+            raise ValueError(msg)
+        elif isinstance(other, pd.Series):
+            # ToDo
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Should this todo still be here?

**Comment by mrocklin** [comment]
This looks good to me.  My suggestions above are minor.

**Comment by sinhrks** [comment]
Thanks for review. Fixed both.

**Pull request merged**